.github/workflows/build_with_current_nightly.yml

```diff
@@ -0,0 +1,55 @@
+name: Build and Test with last nightly everyday @12PM
+
+on:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron:  '00 12 * * *'
+    
+jobs:
+  build_with_last_nightly:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
+    strategy:
+      matrix:
+        device: ["nanos", "nanox", "nanosplus"]
+    steps:
+      - name: update nightly toolchain
+        run: |
+          rustup update nightly
+          rustup default nightly
+          rustup component add rust-src --toolchain nightly
+          cargo ledger setup
+
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          repository: LedgerHQ/app-boilerplate-rust
+          path: app-boilerplate
+
+      - name: Build
+        run: |
+          BUILD_DEVICE_NAME="$(echo ${{ matrix.device }})"
+          BIN_DIR_NAME="$(echo ${{ matrix.device }} | sed 's/nanosplus/nanos2/')"
+          cd app-boilerplate
+          cargo ledger build ${{ matrix.device }}
+          mkdir -p build/${BIN_DIR_NAME}/bin
+          mv ./target/${BUILD_DEVICE_NAME}/release/app-boilerplate-rust ./build/${BIN_DIR_NAME}/bin/app.elf
+      
+      - name: Upload binary artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: "app_elf_binaries"
+          path: app-boilerplate/build/*
+          if-no-files-found: error
+  
+  ragger_tests:
```

Comment:
You will need to run this in the dev-tools image (not the regular app-builder, it doesn't have Ragger nor Speculos)

Reply:
Shall be fixed in [reusable_ragger_tests](https://github.com/LedgerHQ/ledger-app-workflows/blob/master/.github/workflows/reusable_ragger_tests.yml) flow as it uses `ubuntu-latest` nowadays.

Reply:
Weird to use `ubuntu-latest` and then do 

```
      - name: Install tests dependencies
        run: |
          sudo apt-get update && sudo apt-get install -y qemu-user-static tesseract-ocr libtesseract-dev
          pip install -U pip setuptools
          pip install -r "${{ needs.call_get_app_metadata.outputs.pytest_directory }}/requirements.txt"
```

in the reusable workflow... Maybe it's because we want to be closer to a real host environment ? :thinking:

Reply:
Please see https://github.com/LedgerHQ/ledger-app-workflows/pull/56, I do not know why an ubuntu env is used to run test (rather than our dev-tools container)...